qualtran/bloqs/arithmetic/sorting_test.py

```diff
@@ -27,19 +30,54 @@ def test_bitonic_sort(bloq_autotester):
 
 
 def test_comparator_manual():
-    bloq = Comparator(4)
-    assert bloq.t_complexity().t == 88 - 4
-    with pytest.raises(NotImplementedError):
-        bloq.decompose_bloq()
+    bloq = Comparator(2**4)
+    assert bloq.t_complexity().t == 88 - 4 - 7 * 4
```

Comment:
why the -7 * 4?

Reply:
I had some reason to justify this, but can't remember anymore. I _think_ somewhere in the previous t_complexity it assumed CSwap was `t=14`, but the current impl is `t=7`

Reply:
The paper assumes CSWAP is `t=4` using 1 ancilla, right? We should probably open an issue to track that though. Textbook CSWAP should be `t=7`. 


CSWAP to 1 Toffoli can be reduced as follows
<img width="234" alt="image" src="https://github.com/quantumlib/Qualtran/assets/7863287/5255fb0b-c7a5-452e-a6b7-a4bb2d1670d8">


and then the Toffoli can be implemented using 4 T either using a Relative phase toffoli or using an ancilla with AND gate.